core/src/main/scala/kanaloa/reactive/dispatcher/queue/QueueProcessor.scala

```diff
@@ -155,47 +135,37 @@ trait QueueProcessor extends Actor with ActorLogging with MessageScheduler {
  * @param resultChecker
  */
 case class DefaultQueueProcessor(
```

Comment:
One suggestion would be to just have a single class now, `QueueProcessor` instead of the trait and class that we have now.  Since this all ultimately becomes an Actor I don't see much value in the trait, since Actors don't have anything publicly available anyway.